nucleus/payara-modules/requesttracing-core/src/main/java/fish/payara/nucleus/requesttracing/store/ClusteredRequestTraceStore.java

```diff
@@ -54,46 +54,30 @@
  */
 public class ClusteredRequestTraceStore implements RequestTraceStoreInterface, Serializable {
 
-    private final String instanceId;
-    private final MultiMap<String, RequestTrace> store;
+    private final Map<UUID, RequestTrace> store;
     private int maxStoreSize;
-    
+
     private final TraceStorageStrategy strategy;
 
-    ClusteredRequestTraceStore(MultiMap<String, RequestTrace> store, String instanceId, TraceStorageStrategy strategy) {
+    ClusteredRequestTraceStore(Map<UUID, RequestTrace> store, TraceStorageStrategy strategy) {
```

Comment:
I have no clue why the `instanceId` came in but it does not make sense as it would effectively give each instance its own store just that it is stored distributed instead of  in local memory. I believe the goal here was to have all instances share a common distributed store. This is best done by a simpler distributed data structure. I opted for a `Map` where the keys are the trace IDs and the values are the traces.  The instances will all use the same map as it has a constant name within the cluster. A small side benefit is that monitoring will have statistics available on the map.

Reply:
This can be a topic for Monday meeting since it also has consequences for the UI. If there is a store per instance and given we make all of them accessible in the UI you get instances times max-size traces which sounds like could be quite a number at times.

Reply:
I was thinking about this a bit more and to me it appears less and less desirable to have a non-shared  store. While instances theoretically do have access to the data of all instances the reality is that asking the store only makes the local data visible. As a consequence when using the admin GUI you can never look into anything not deployed on the DAS which doesn't feel that useful. Say we would allow to get data from other instances there would still be the problem that any tool using request tracing data had to make this decision or forward it and there is the risk of configuration not being what the user wanted for something were it is quite hard to verify that a certain area is covered. I'd think for something that should warn you about something that you might not be able to trigger to verify correct configuration there should not be such a configuration so you can be sure that the tool will warn you